### PR TITLE
Add some documentation about SpECTRE's use of Travis

### LINKED ISSUE #201: document how to clear the Travis CI cache
If Travis CI changes something like their architecture or OS, our builds may fail because of caching.  In order to fix the build, someone with admin privileges needs to go to 'More options' -> 'Caches' and delete the appropriate cache.

### PR DESCRIPTION
**Pull request by kidder**
## Proposed changes

Add documentation about Travis

Fix #201 

### Types of changes:

- [ ] Bugfix
- [x] New feature

### Component:

- [ ] Code
- [x] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] Follows [code review guidelines](https://sxs-collaboration.github.io/spectre/code_review_guide.html)
- [ ] Code has documentation and unit tests
- [ ] Private member variables have a trailing underscore
- [ ] Do not use [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation), e.g. `double* pd_blah` is bad
- [ ] Header order:
  1. hpp corresponding to cpp (only in cpp files)
  2. Blank line (only in cpp files)
  3. STL and externals (in alphabetical order)
  4. Blank line
  5. SpECTRE includes (in alphabetical order)
- [ ] File lists in CMake are alphabetical
- [ ] Correct `noexcept` specification for functions (if unsure, mark `noexcept`)
- [ ] Mark objects `const` whenever possible
- [ ] Almost always `auto`, except with expression templates, i.e. `DataVector`
- [ ] All commits for performance changes provide quantitative evidence and the tests used to obtain said evidence.
- [ ] Make sure error messages are helpful, e.g. "The number of grid points in the matrix 'F' is not the same as the number of grid points in the determinant."
- [ ] Prefix commits addressing PR requests with `fixup`

### CODE REVIEW

**Comment by nilsdeppe** [comment]
Here's the built documentation: http://nilsdeppe.com/spectre/pr_533/

File: docs/DevGuide/Travis.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+\endcond
+
+# Travis CI {#travis_guide}
+
+SpECTRE uses
+[TravisCI](https://travis-ci.org/sxs-collaboration/spectre) for
+testing the code.  Multiple build jobs (described below) are launched
+each time a pull request is submitted or updated.  In addition, a
+daily cron job is run.  Travis will also launch these build jobs each
+time you push to a branch on your fork of SpECTRE.  For pull requests,
```
Comment (wthrowe) [inline docs/DevGuide/Travis.md]: It will?  How can I see the results?

File: docs/DevGuide/Travis.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  - wip (for work in progress)
+  - fixme
+  - deleteme
+  - rebaseme
+  - testing
+  - rebase
+* CHECK_FILES runs the script `tools/CheckFiles.sh` (which also runs the script
+`tools/FileTestDefs.sh`) and fails the build if any of the following checks are
+true:
+  - A `c++` file (i.e. `*.hpp` or `*.cpp` file) contains a line over 80
```
Comment (wthrowe) [inline docs/DevGuide/Travis.md]: This should also include `*.tpp`, and I will submit a pull request momentarily to make that actually true.

**Review by wthrowe — CHANGES_REQUESTED** [review]

File: docs/DevGuide/Travis.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+# Travis CI {#travis_guide}
+
+SpECTRE uses
+[TravisCI](https://travis-ci.org/sxs-collaboration/spectre) for
+testing the code.  Multiple build jobs (described below) are launched
+each time a pull request is submitted or updated.  In addition, a
+daily cron job is run.  Travis will also launch these build jobs each
+time you push to a branch on your fork of SpECTRE.  For pull requests,
+you can see view the Travis CI build by clicking on `Details` next to
```
Comment (kidder) [inline docs/DevGuide/Travis.md]: see or view

File: docs/DevGuide/Travis.md (hunk 1 above)
Comment (kidder) [reply docs/DevGuide/Travis.md]: Apparently I set it up this way; I will document how (if anyone can) or remove the comment (if it required administrator priv)

File: docs/DevGuide/Travis.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+the `Travis build failed` or `Travis build succeeded` in the section
+for `All checks have failed` or `All checks have passed` towards the
+bottom of the pull request.  You can login to Travis using your GitHub
+credentials in order to cancel your builds and enable additional
```
Comment (kidder) [inline docs/DevGuide/Travis.md]: remove and enable... (requires super admin privs)

File: docs/DevGuide/Travis.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+options.
+
+## What is tested
+
+The Travis report lists the build jobs which will each have either a
+green check mark if it passes, or a red `X` if it has failed.
```
Comment (kidder) [inline docs/DevGuide/Travis.md]: or a yellow binary if the build is in progress

File: docs/DevGuide/Travis.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  - A `*.hpp` file is missing `pragma once`
+  - A file does not end with a newline
+  - A `c++` file includes any of the following headers
+    * `<iostream>`  (useless when running in parallel)
+    * `Utilities/TmplDebugging.hpp`  (used only for debugging)
+  - A `c++` file contains a `namespace` ending in `_details` (use `_detail`)
+  - A `c++` file contains a `struct TD` or `class TD` (used only for debugging)
+  - A `c++` file contains `std::enable_if` (use `Requires` instead)
+  - A `c++` file contains additional text after `/*!` (does not render correctly
+  in doxygen)
```
Comment (kidder) [inline docs/DevGuide/Travis.md:62]: capitalize

File: docs/DevGuide/Travis.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+`clang-tidy` on all files which were modified.  This is done for both `Release`
+and `Debug` builds.
+* RUN_CPPCHECK runs `make cppcheck`.  This is done for both `Release` and
+`Debug` builds.
+* TEST_CHECK_FILES runs `tools/CheckFiles.sh --test` which tests the checks
+performed in the CHECK_FILES build.
+* The other builds compile the code and run the tests for both
+`Release` and `Debug` builds, for the `gcc` and `clang` compilers
+using a Linux OS, and the `AppleClang` compiler for `OS X`.
+* The `gcc Debug` build will fail if there are `doxygen` warnings.
```
Comment (kidder) [inline docs/DevGuide/Travis.md:77]: capitalize

File: docs/DevGuide/Travis.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+* The `gcc Debug` build runs code coverage for each Travis build.
+
+## Troubleshooting
+
+* Occasionally, a build job will fail because of a problem with
+Travis (e.g. it times out).  Clicking on the icon on the far right of
+the build job will restart it.  All build jobs can be restarted by
+clicking on the `Restart build` button to the right of the pull
+request description.
+* Travis caches some things between builds.  Occasionally this may
```
Comment (kidder) [inline docs/DevGuide/Travis.md]: only available to admins

File: docs/DevGuide/Travis.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+* The `gcc Debug` build will fail if there are `doxygen` warnings.
+
+## Travis setup
+
+* The `gcc Debug` build runs code coverage for each Travis build.
+
+## Troubleshooting
+
+* Occasionally, a build job will fail because of a problem with
+Travis (e.g. it times out).  Clicking on the icon on the far right of
```
Comment (kidder) [inline docs/DevGuide/Travis.md]: circular arrow icon

File: docs/DevGuide/Travis.md (hunk 1 above)
Comment (kidder) [reply docs/DevGuide/Travis.md]: Go to Travis, click on your name in the upper right corner, and then towards the bottom, enable testing on your fork of spectre

**Review by nilsdeppe — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Pull request merged**